tensorbay/opendataset/SVHN/loader.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name, missing-module-docstring
+
+import os
+from typing import Dict, List
+
+from ...dataset import Data, Dataset
+from ...exception import ModuleImportError
+from ...label import LabeledBox2D
+
+_SVHN_SEGMENTS = ("extra", "test", "train")
+
+DATASET_NAME = "SVHN"
+
+
+def SVHN(path: str) -> Dataset:
+    """Dataloader of SVHN open dataset.
+
+    .. SVHN: http://ufldl.stanford.edu/housenumbers/
+
+    The file structure should be like::
+
+        <path>
+            Cropped/
+                extra_32x32.mat
+                test_32x32.mat
+                train_32x32.mat
+            FullNumbers/
+                extra/
+                    116507.png
+                    116508.png
+                    ...
+                    digitStruct.mat
+                    see_bboxes.m
+                test/
+                train/
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class: `~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    try:
+        from h5py import Dataset as h5Dataset
+        from h5py import File  # pylint: disable=import-outside-toplevel
+    except ModuleNotFoundError as error:
+        raise ModuleImportError(error.name) from error  # type: ignore[arg-type]
+
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in _SVHN_SEGMENTS:
+        segment = dataset.create_segment(f"FullNumbers-{segment_name}")
```

Comment:
why we add prefix `FullNumber` to the segment name?
can we just use test, train and extra as segment name?